### DISCUSSION TITLE
How to debounce the linter?

### DISCUSSION THREAD

**Discussion by coolaj86** [w0rp/ale#3875] [Q&A - Ask for help with problems]
I just started using `tsserver` as my linter for JSDoc + JavaScript, and it absolutely make vim unusable. There are 30 second pauses every time I edit an `@param`... or just at random while typing.

Is there a way to debounce a particular linter so that it only kicks in on save or after 5s without keystrokes or something like that?

**Comment by hsanson** [comment]
Does g:ale_lint_on_save help?

> :h g:ale_lint_on_save

For overview of how ALE trigger lints and options to control it check:

> :h ale-lint

**Reply by coolaj86** [reply]
I got `E149: Sorry, no help for ale-lint`.

How do I build the `ale-lint` help, or where do I view it online?

(got that for both, actually)

**Reply by coolaj86** [reply]
Ideally I would like to only enable that option for projects where `tsconfig.json` can be found in some parent directory, or perhaps only in JavaScript projects, but not all projects.

Or perhaps with a per-project `.vimrc`

`~/.vimrc`:

```vim
set exrc
set secure
```

`~/my-project/.vimrc`:

```vim
let g:ale_lint_on_save = 1
```
